dfttopif/parsers/vasp.py

```diff
@@ -357,3 +357,15 @@ def get_dos(self):
             # Convert to property
             return Property(scalars=dos, units='number of states per unit cell',
                             conditions=Value(name='energy', scalars=energy, units='eV'))
+
+    def get_total_magnetization(self):
+        file_path = os.path.join(self._directory, "OUTCAR")
+        if not os.path.isfile(file_path):
+            return None
+        parser = OutcarParser()
+        with open(file_path, "r") as fp:
```

Comment:
Do we really read the entire outcar every time that we need to extract a property?

Reply:
Not every, but most.  I can run this parser through just once, but the rest of the code (which for the most part doesn't use dftparse) will still re-read it.